internal/cli/up.go

```diff
@@ -100,6 +103,13 @@ local resources--i.e. those using serve_cmd--are terminated when you exit Tilt.
 		c.hudFlagExplicitlySet = cmd.Flag("hud").Changed
 	}
 
+	//since the --watch flag isn't really useful and has been made obselete by tilt ci, print a deprecation warning.
+	cmd.Flag("watch").Deprecated = "it will be removed in future releases."
+	//if a deprecation message exists, then watch was explicitly set by the user, so mark the watchFlagExplicitlySet var as true for
+	//logging the message on the web UI and TUI
+	if len(cmd.Flag("watch").Deprecated) > 0 {
```

Comment:
i don't think this is doing what you expect...this will always be true, because you just set it above.

we probably shouldn't use the `Deprecated` field at all, because we don't want the flag library to do its own printing?

Reply:
Ohh right, I assumed that the `Deprecated` property is only invoked when the user explicitly uses the flag in the command line, and that it would just be an empty string otherwise. Turns out that is not the case, my bad. 

Yeah, I think I'll look out for another way to do this because of the flag library printing. 

(Also, apologies for the late reply, it was getting late yesterday and I was a bit tired)

Reply:
I think it's good now! I got too hung up on utilizing the `Deprecated` field (since I thought that was the best way to get about it) but turns out, it was just a matter of adding another line under the existing logic: 
`cmd.PreRun = func(cmd *cobra.Command, args []string) {
		c.hudFlagExplicitlySet = cmd.Flag("hud").Changed
		c.watchFlagExplicitlySet = cmd.Flag("watch").Changed
	}

`
Its printing the deprecation warning on the logs both on the TUI and the Web UI properly now (and doesn't print the message when the flag isn't present) 

![Screenshot from 2020-06-12 10-16-47](https://user-images.githubusercontent.com/35721712/84471227-f7758480-ac95-11ea-8afa-bcbd04d5a62b.png)

![Screenshot from 2020-06-12 10-17-14](https://user-images.githubusercontent.com/35721712/84471234-f9d7de80-ac95-11ea-9271-1b2b94c12454.png)